fiftyone/core/stages.py

```
@@ -389,6 +448,133 @@ class ViewStageError(Exception):
+    """
+    envelopes = {}
+    for name, value in expressions.items():
+        if isinstance(value, foe.ViewExpression) and foea.is_reconstructible(
+            value
+        ):
+            envelopes[name] = foea.to_envelope(value)
+
+    return envelopes
+
+
+def _decode_expressions(value):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_decode_expressions`

(ANN202)